scripts/analyze.py

```
@@ -0,0 +1,462 @@
+            print(f"    Special:    {cats.get('special_pct', 0)}%")
+            print(f"\n  Patterns (sample):")
+            pat = comp.get('patterns', {})
+            print(f"    Date-like:      {pat.get('date_like_pct', 0)}%")
+            print(f"    Keyboard walk:  {pat.get('keyboard_walk_pct', 0)}%")
+            print(f"    Repeated:       {pat.get('repeated_pattern_pct', 0)}%")
+        sys.exit(0)
+
+    # Cross-reference
+    if args.cross_reference:
+        xref = cross_reference(filepaths)
+        print(f"\nCross-Reference Results:")
```

Ruff (0.15.17)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)